### PR TITLE
[IOTDB-4504] Add BackwardSort algorithm for disordered time series

### PR DESCRIPTION
**Pull request by LittleHealth**
## design and implement BackwardSort algorithm for disordered time series

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java [hunk 1]
```diff
@@ -81,10 +81,42 @@ public abstract class AlignedTVList extends TVList {
   public static AlignedTVList newAlignedList(List<TSDataType> dataTypes) {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickAlignedTVList(dataTypes);
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackAlignedTVList(dataTypes);
     }
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java:86]: use switch?

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/BinaryTVList.java [hunk 2]
```diff
@@ -52,6 +52,8 @@ public abstract class BinaryTVList extends TVList {
   public static BinaryTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickBinaryTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackBinaryTVList();
     }
     return new TimBinaryTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/BinaryTVList.java:58]: Use switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/BooleanTVList.java [hunk 3]
```diff
@@ -51,6 +51,8 @@ public abstract class BooleanTVList extends TVList {
   public static BooleanTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickBooleanTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackBooleanTVList();
     }
     return new TimBooleanTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/BooleanTVList.java:57]: Use switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/DoubleTVList.java [hunk 4]
```diff
@@ -51,6 +51,8 @@ public abstract class DoubleTVList extends TVList {
   public static DoubleTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickDoubleTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackDoubleTVList();
     }
     return new TimDoubleTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/DoubleTVList.java:57]: Use switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/FloatTVList.java [hunk 5]
```diff
@@ -51,6 +51,8 @@ public abstract class FloatTVList extends TVList {
   public static FloatTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickFloatTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackFloatTVList();
     }
     return new TimFloatTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/FloatTVList.java:57]: Use switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/IntTVList.java [hunk 6]
```diff
@@ -50,6 +50,8 @@ public abstract class IntTVList extends TVList {
   public static IntTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickIntTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackIntTVList();
     }
     return new TimIntTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/IntTVList.java:56]: Use switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/LongTVList.java [hunk 7]
```diff
@@ -50,6 +50,8 @@ public abstract class LongTVList extends TVList {
   public static LongTVList newList() {
     if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.QUICK) {
       return new QuickLongTVList();
+    } else if (TVLIST_SORT_ALGORITHM == TVListSortAlgorithm.BACKWARD) {
+      return new BackLongTVList();
     }
     return new TimLongTVList();
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/utils/datastructure/LongTVList.java:56]: switch

File: server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java (hunk 1 above)
Comment (HTHou) [reply server/src/main/java/org/apache/iotdb/db/utils/datastructure/AlignedTVList.java:86]: there is one left

**Review by HeimingZ — APPROVED** [review]
LGTM :)

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**